### PR TITLE
cpu: native: work around shared errno in _native_lpm_sleep

### LINKED ISSUE #3222: ng_nativenet crashes when hammered
using "ping6 -f <ll-addr>%tapX" from two threads on linux to riot at the same time, I can reliably crash native:

<pre>
kernel_init(): This is RIOT! (Version: 2014.12-1938-g910a-booze)
kernel_init(): jumping into first task...
UART0 thread started.
uart0_init() [OK]
RIOT network stack example application
All up, running the shell now
> /home/kaspar/src/riot.8/examples/ng_networking/bin/native/ng_networking.elf: lpm_set(): select(): Resource temporarily unavailable
[kaspar@booze ng_networking (master)]$
</pre>

Weird thing is that it's crashing in ```lpm_set()``` with "Resource temporarily unavailable", which points to EAGAIN in errno, which should have been handled before. Looks like a race.

@LudwigOrtmann, does native share errno between all threads, like "normal" RIOT?

### PR DESCRIPTION
**Pull request by kaspar030**
Seems to fix  #3222.

### CODE REVIEW

File: cpu/native/lpm_cpu.c [hunk 1]
```diff
@@ -55,6 +55,7 @@ void _native_lpm_sleep(void)
 
     _native_in_syscall++; // no switching here
     nfds = real_select(nfds, &_native_rfds, NULL, NULL, NULL);
+    int _errno = errno;
```
Comment (Kijewski) [inline cpu/native/lpm_cpu.c]: Is this variable somewhere written to or is it deterministically 0 for ever?

Comment (kaspar030) [inline cpu/native/lpm_cpu.c]: which one?

**Comment by PeterKietzmann** [comment]
I did several experiments and can confirm that the error described in #3222 didn't occur any more with this PR. I don't think I'm in the right position to decide if this workaround is legit.

**Comment by PeterKietzmann** [comment]
Anyone dares to merge this?

**Comment by kaspar030** [comment]
Yes, it makes native a lot more stable for me.

**Pull request merged**